### PR TITLE
Ref #4879: VPN churn improvements - VerifyReceiptData fields with isBillingRetry

### LINKED ISSUE #4879: VPN Billing Retry Changes
Details can be followed in https://github.com/brave/brave-ios/pull/8040

### PR DESCRIPTION
**Pull request by soner-yuksel**
## Summary of Changes

Second part Pr for VPN Churn improvement. After https://github.com/brave/brave-ios/pull/7670 added Pop-over types and new style of displaying the pop-overs.

This PR is adding fixes for proper priority for Callouts and method for displaying update billing pop over.

In addition main part is implementation of `verifyReceiptData` function which is right now only used in AppLaunch which can retrieve all details of receipt.

The tests are added for receipt field different variation.

There will be a 3 rd Pr after this to combine menu changes and popover UX depending on this PR. But this Pr can be merged when review is finished and application works as expected with the change of verifyReceipt data functions

This pull request fixes #4879

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Test Plan:
N/A

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Sources/BraveVPN/BraveVPN.swift [hunk 1]
```diff
@@ -80,9 +98,8 @@ public class BraveVPN {
       return
     }
     
-    // We validate the current receipt at the start to know if the subscription has expirerd.
-    BraveVPN.validateReceipt() { expired in
-      if expired == true {
+    BraveVPN.validateReceiptData() { receiptResponse in
```
Comment (soner-yuksel) [inline Sources/BraveVPN/BraveVPN.swift:101]: New validate method is used for initialization.

File: Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+Callout.swift [hunk 2]
```diff
@@ -244,11 +257,24 @@ extension BrowserViewController {
...
   
-  private func presentVPNChurnPromoCallout(for type: VPNChurnPromoType) {
+  private func presentVPNUpdateBillingCallout(skipSafeGuards: Bool = false) {
+    if !skipSafeGuards {
+      // TODO: Condition
+      return
+    }
+    
+    presentVPNChurnPromoCallout(for: .updateBillingExpired) {
+      // TODO: Action
```
Comment (iccub) [inline Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+Callout.swift:267]: Let's open iOS 'manager subscription' url here.
Unless you find a better url to directly update payment info

File: Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+Callout.swift [hunk 3]
```diff
@@ -244,11 +257,24 @@ extension BrowserViewController {
     present(popup, animated: false)
   }
   
-  private func presentVPNChurnPromoCallout(for type: VPNChurnPromoType) {
+  private func presentVPNUpdateBillingCallout(skipSafeGuards: Bool = false) {
+    if !skipSafeGuards {
+      // TODO: Condition
```
Comment (iccub) [inline Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+Callout.swift:262]: Do we still have to decide on something here?

**Review by iccub — CHANGES_REQUESTED** [review]

File: Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+Callout.swift (hunk 3 above)
Comment (soner-yuksel) [reply Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+Callout.swift:262]: As I mentioned in previous comment I want to address action and condition to present pop up in next PR along with other UI changes. We have to decide when to show this pop over right there :D

File: Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+Callout.swift (hunk 2 above)
Comment (soner-yuksel) [reply Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+Callout.swift:267]: I will take care of action part and condition of presentation in upcoming PR.

**Review by iccub — APPROVED** [review]

**Pull request merged**